services/web/server/src/simcore_service_webserver/api_keys/_repository.py

```diff
@@ -73,62 +35,90 @@ async def get_or_create_api_key(
-        insert_stmt = (
+        stmt = (
             pg_insert(api_keys)
             .values(
                 display_name=display_name,
                 user_id=user_id,
                 product_name=product_name,
                 api_key=api_key,
-                api_secret=api_secret,
+                api_secret=_hash_secret(api_secret),
                 expires_at=(sa.func.now() + expiration) if expiration else None,
             )
             .on_conflict_do_update(
                 index_elements=["user_id", "display_name"],
                 set_={
-                    "product_name": product_name
-                },  # dummy enable returning since on_conflict_do_nothing returns None
-                # NOTE: use this entry for reference counting in https://github.com/ITISFoundation/osparc-simcore/issues/5875
+                    "api_key": api_key,
+                    "api_secret": _hash_secret(api_secret),
+                    "expires_at": (sa.func.now() + expiration) if expiration else None,
+                },
             )
             .returning(api_keys)
         )
 
-        result = await conn.stream(insert_stmt)
-        row = await result.first()
-        assert row  # nosec
+        result = await conn.stream(stmt)
+        row = await result.one()
 
         return ApiKey(
             id=f"{row.id}",  # NOTE See: https://github.com/ITISFoundation/osparc-simcore/issues/6919
-            display_name=row.display_name,
-            expiration=row.expires_at,
-            api_key=row.api_key,
-            api_secret=row.api_secret,
+            display_name=display_name,
+            expiration=expiration,
+            api_key=api_key,
+            api_secret=api_secret,
         )
 
 
-async def list_api_keys(
+async def delete_api_key(
     app: web.Application,
     connection: AsyncConnection | None = None,
     *,
     user_id: UserID,
     product_name: ProductName,
-) -> list[ApiKey]:
-    async with pass_or_acquire_connection(get_asyncpg_engine(app), connection) as conn:
-        stmt = sa.select(api_keys.c.id, api_keys.c.display_name).where(
-            (api_keys.c.user_id == user_id) & (api_keys.c.product_name == product_name)
+    api_key_id: str,
+) -> None:
+    async with transaction_context(get_asyncpg_engine(app), connection) as conn:
+        stmt = api_keys.delete().where(
+            (
+                api_keys.c.id == int(api_key_id)
+            )  # NOTE See: https://github.com/ITISFoundation/osparc-simcore/issues/6919
+            & (api_keys.c.user_id == user_id)
+            & (api_keys.c.product_name == product_name)
         )
+        await conn.execute(stmt)
 
-        result = await conn.stream(stmt)
-        rows = [row async for row in result]
 
-        return [
-            ApiKey(
-                id=f"{row.id}",  # NOTE See: https://github.com/ITISFoundation/osparc-simcore/issues/6919
-                display_name=row.display_name,
+async def delete_api_key_by_key(
```

Comment:
MINOR: haha the names get very funny. I wonder we should do a small rename as `api_auth` . See my comments above

Reply:
Nice idea, but not the purpose of **THIS** PR.